packages/node-core/src/indexer/project.service.ts

```diff
@@ -24,7 +24,7 @@ class NotInitError extends Error {
   }
 }
 
-export abstract class BaseProjectService<DS extends {startBlock?: number}> implements IProjectService<DS> {
+export abstract class BaseProjectService<A, SA, B, DS extends {startBlock?: number}> implements IProjectService<DS> {
```

Comment:
```suggestion
export abstract class BaseProjectService<Api extends IApi, DS extends {startBlock?: number}> implements IProjectService<DS> {
```

Reply:
I think we should let it extend ApiService because it accesses `networkMeta`